Add FastxRecord to the docs

nh13 opened the pull request.

nh13 committed:
Add FastxRecord to the docs

Add FastxRecord to API documentation and improve its docstring with
proper Sphinx-style documentation including attributes, parameters,
and usage examples.

jmarshall commented:
When we add a previously omitted class to _api.rst_, we suddenly find out whether its docstrings render well as reStructuredText or provide useful information at all. In this case, some of it's not great:

> `name`
>  &nbsp;&nbsp;&nbsp;&nbsp;unicode
>  &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;**Type:** name

These appear to be the default rendering for `cdef public str` from the _.pxd_ file. (These are the only `public str` items in pysam.) “Unicode” is irrelevant as we're Python 3 only, and “Type” is just wrong.


Perhaps we can add some docstrings to the _.pyx_ file to override this rendering and provide some meaningful information for these properties…

jmarshall committed:
Move the attribute documentation to `cdef public ...` in the .pxd file,
where Cython will use it to populate these attributes' docstrings.

Remove embedsignature=True as it caused these to get bad type information.
This unembedding removes the `self` parameter from the documentation of
seven existing methods in libcfaidx.pyx, but this arguably improves them.

Add/improve the documentation of the other Python-visible FastxRecord
methods. Add a __repr__() method for ease of observation.

nh13 commented:
Thanks @jmarshall!
